--- jlab.sh	2018-11-19 19:38:30.000000000 -0600
+++ jlab.sh	2020-05-10 20:25:58.215883143 -0500
@@ -16,7 +16,16 @@
 : ${JLAB_VERSION=2.3}
 
 TMPF=`mktemp`
-tcsh ${TCSH_ARG} -c "source ${JLAB_ROOT}/${JLAB_VERSION}/ce/jlab.csh $1; printenv" | perl -e '
+# start with empty environment
+env -i tcsh ${TCSH_ARG} -c "
+    set JLAB_ROOT=${JLAB_ROOT} ;
+    set JLAB_VERSION=${JLAB_VERSION} ;
+    set PATH=${PATH} ;
+    set LD_LIBRARY_PATH=${LD_LIBRARY_PATH} ;
+    set DYLD_LIBRARY_PATH=${DYLD_LIBRARY_PATH} ;
+    set home=${HOME} ;
+    source ${JLAB_ROOT}/${JLAB_VERSION}/ce/jlab.csh $1 ;
+    printenv" | perl -e '
   my %IGNORE;
   ## Add variables to ignore to the space-delimited list below
   foreach( qw(
@@ -27,7 +36,7 @@
     chomp;
 
     ## Look for and treat variables from printenv here
-    if(/(.*?)=(.*)/) {
+    if(/^([_a-zA-Z0-9]*?)=(.*)$/) {
       my $var = $1;
       my $arg = $2;
 
